src/js/files/components/UploadItem.tsx

```diff
@@ -37,16 +42,35 @@ const UploadItemTitle = styled.div`
     }
 `;
 
+/**
+ * Styled component for the name of the upload item.
+ */
 const UploadItemName = styled.span`
     font-weight: ${getFontWeight("thick")};
 `;
 
-export const UploadItem = ({ name, progress, size, failed, localId, onRemove }) => {
+/**
+ * Props definition for the UploadItem component.
+ */
+interface UploadItemProps {
+    name: string;
+    progress: number;
+    size: number;
+    failed: boolean;
+    localId: string;
+    onRemove: (localId: string) => void;
+}
```

Comment:
```suggestion
type UploadItemProps = {
    name: string;
    progress: number;
    size: number;
    failed: boolean;
    localId: string;
    onRemove: (localId: string) => void;
}
```

Differences in functionality between type and interface are minimal, but the two don't work together well

~~Not blocking (aka will merge without): Try adding docstrings for the props themselves. Not all of them need description, but the use case for something like `localId` may not be obvious at first look.~~

Outdated since last push. Thanks for jumping on that! That said the format we use is typically todo the docs right next to the type:

interface UploadItemProps {
/** <Relevant docstring>*/
    name: string;
    ...ect
}